rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/BuildTaskEndpointImpl.java

```diff
@@ -113,8 +167,194 @@ public Response buildTaskCompleted(String buildId, BuildResultRest buildResult)
     }
 
     @Override
+    @Deprecated // TODO remove once fully migrated to Rex
     public Response buildTaskCompletedJson(String buildId, BuildResultRest buildResult)
             throws org.jboss.pnc.facade.validation.InvalidEntityException {
         return buildTaskCompleted(buildId, buildResult);
     }
+
+    @Override
+    @Transactional
+    public Response buildTaskNotification(@NotBlank String buildId, @NotNull NotificationRequest notification)
+            throws InvalidEntityException {
+        logger.debug("Received transition notification for Build '{}'", buildId);
+
+        if (!buildId.equals(notification.getTask().getName())) {
```

Comment:
Rex Task should match buildID. Somewhere along the code I map RexTask to BuildTask stuff and it would be good to have the endpoint buildID match the record that gets saved. I'll comment it up.